### PR TITLE
Fix re-uploads of same wdl file: GAWB-2547

### PR DESCRIPTION
**Pull request by rushtong**
Addresses 
https://broadinstitute.atlassian.net/browse/GAWB-2547

Use the same approach as https://github.com/broadinstitute/firecloud-ui/pull/788 for the wdl uploader.

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by davidangb — APPROVED** [review]
:+1: works for me!

File: src/cljs/main/broadfcui/page/method_repo/create_method.cljs [hunk 1]
```diff
@@ -37,6 +37,9 @@
   {:component-will-mount
    (fn [{:keys [props locals]}]
      (swap! locals assoc :info (build-info props)))
+   :get-initial-state
+   (fn []
+     {:file-input-key (gensym "wdl-uploader-")})
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/method_repo/create_method.cljs]: Do we need this? Is it a problem if there's no key the first time, and then one gets created afterwards? Not a big deal either way.

File: src/cljs/main/broadfcui/page/method_repo/create_method.cljs [hunk 2]
```diff
@@ -79,15 +82,23 @@
                                     :defaultValue (:documentation info)
                                     :style {:width "100%"}
                                     :rows 5})
-
-           [:input {:type "file" :ref "wdl-uploader" :style {:display "none"}
+           ;; This key is changed every time a file is selected causing React to completely replace the
+           ;; element. Otherwise, if a user selects the same file (even after having modified it), the
+           ;; browser will not fire the onChange event.
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/method_repo/create_method.cljs:84]: Excellent explanation.

**Review by zarsky-broad — APPROVED** [review]
One possible nit, otherwise looks good.

File: src/cljs/main/broadfcui/page/method_repo/create_method.cljs (hunk 2 above)
Comment (rushtong) [reply src/cljs/main/broadfcui/page/method_repo/create_method.cljs:84]: You can thank Ursa for that.

File: src/cljs/main/broadfcui/page/method_repo/create_method.cljs (hunk 1 above)
Comment (rushtong) [reply src/cljs/main/broadfcui/page/method_repo/create_method.cljs]: Will double check - but possibly not.

Comment (rushtong) [reply src/cljs/main/broadfcui/page/method_repo/create_method.cljs]: It does work without this - will remove.

**Comment by zarsky-broad** [comment]
Awesome.

**Pull request merged**